components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtils.java

```diff
@@ -122,11 +122,12 @@ public static DCRMClientException generateClientException(DCRMConstants.ErrorMes
     public static boolean isRegexValidated(String applicationName) {
 
         String spValidatorRegex = DCRConstants.APP_NAME_VALIDATING_REGEX;
-        Pattern regexPattern = Pattern.compile(spValidatorRegex);
-        if (regexPattern.matcher(applicationName).matches()) {
-            return true;
-        } else {
-            return false;
-        }
+        return isRegexValidated(applicationName, spValidatorRegex);
+    }
+
+    public static boolean isRegexValidated(String providedString, String regex) {
+
+        Pattern regexPattern = Pattern.compile(regex);
```

Comment:
Keep the compiled pattern as a constant/class variable as the pattern is constant here. Pattern compiling can be costly if called frequently.

Reply:
This method is not used anymore

Reply:
If these methods are not supposed to be used anymore, shall we add deprecated annotation.